src/pages/MentorProfile/MentorProfile.component.tsx

```diff
@@ -101,11 +101,51 @@ const MentorProfile: React.FC = () => {
                 <ShareIcon />
               </span>
             </div>
-            <p className="text-sm">
-              {mentor?.application.position},{' '}
-              <span className="text-gray-500">
-                {mentor?.application.institution}
-              </span>
+            <p className="text-md flex flex-wrap items-center">
+              {mentor?.application.noOfMentees && mentor.mentees ? (
+                <span
+                  className={`inline-block text-sm font-medium px-3 py-1 rounded-full mt-2 sm:mt-0 mr-2 ${
+                    Math.max(
+                      0,
+                      mentor.application.noOfMentees -
+                        mentor.mentees.filter(
+                          (mentee) =>
+                            mentee.state === ApplicationStatus.APPROVED
+                        ).length
+                    ) === 0
+                      ? 'bg-gray-100 text-gray-800'
+                      : 'bg-green-100 text-green-800'
+                  }`}
+                >
+                  {Math.max(
+                    0,
+                    mentor.application.noOfMentees -
```

Comment:
Create a variable for this. 

```suggestion
const noOfApprovedMentees = mentor.mentees.filter(
                        (mentee) => mentee.state === ApplicationStatus.APPROVED
                      ).length
```

Reply:
```
// Calculate the number of approved mentees once
const approvedMenteesCount = mentor.mentees.filter(
  (mentee) => mentee.state === ApplicationStatus.APPROVED
).length;

// Calculate available slots
const availableSlots = Math.max(
  0,
  mentor.application.noOfMentees - approvedMenteesCount
);

// Render the result
<>
  {availableSlots}{' '}
  {availableSlots <= 1 ? 'Slot' : 'Slots'}{' '}
  Available
</>
```